tests/RobotFramework/SETUP.md

```diff
@@ -0,0 +1,211 @@
+# Integration testing
+
+## Overview
+
+The testing framework is written to support running on multiple targets either using docker container or using an SSH enabled device. The framework offers a simple way to setup a test using the desired adapter (e.g. `docker` or `ssh`). These adapters is called a `device adapter`. They are written to be device agnostic so they allow for maximum flexibility to write your tests to adapter to your needs.
+
+The default device adapter is `docker`, and the primary support will be for `docker` at first, and the agnostic testing interface will be extended gradually. The `docker` device adapter was chosen as the primary adapter as for the following reasons:
+
+* No hardware dependencies (you can spawn a container pretty much anywhere even in CI/CD)
+* Supports complex testing of network interruptions as you can disconnect the device from the network but still control it via the docker API
+* Have a fresh container each time (reduce side-effects from unrelated tests)
+
+There are obvious drawbacks to the points above, however once the other adapters (`ssh`) are ready it will also supporting running 90% of the same tests on a different target with zero code changes.
+
+## Device adapters
+
+### Docker adapter
+
+The docker adapter will spawn a new container on demand which can be used to run the tests on. The container provides a simple/clean device to run all of the tests, and can be destroyed afterwards. This makes it very convenient to run your tests as it does not require an external hardware.
+
+The general test suite flow is as follows:
+
+1. Create a device
+2. Bootstrap the device (register it with the cloud)
+3. Run tests (including assertions)
+4. Collect device diagnostics (logs, cloud digital twin etc. for post analysis)
+5. Cleanup device artifacts (e.g. delete uploaded certificate, cloud digital twin etc.)
+6. Destroy the device
+
+### SSH adapter
+
+The ssh adapter uses an existing device and uses a SSH connect to run the test suite against it. In this setup, you are responsible for providing a device, container, server before the test can start.
+
+The general test suit flow is very similar to the above [Docker adapter](./SETUP.md#docker-adapter) flow, however the device creation and destroy steps are skipped.
+
+# Setup
+
+## Pre-requisites
+
+Before you can run the tests you need to install the pre-requisites:
+
+* docker
+* python3 (>=3.10)
+* pip3
+* nodejs (>=18)
```

Comment:
Why is nodejs required?

Reply:
Nodejs is still used for some parts of the tutorial settings. I guess it would be technically an optional dependency.

So I can change the working regarding it.

Reply:
> I guess it would be technically an optional dependency.

Don't spend too much time on that. I will try without myself.

Reply:
I removed the note about nodejs, as is currently for an semi-automated test, but I would like to also change it as browser testing in notoriously unreliable and it does not bring much value imho.